### ISSUE TITLE
Allow opting out of "Should Not Happen in non gmail websites" error

### ISSUE THREAD

**Issue by nirvparekh** [InboxSDK/InboxSDK#873]
**Describe the bug**
When we open websites other than https://mail.google.com, it shows error in browser developer console like the screenshot below

**To Reproduce**

1. configure pageWorld: '@inboxsdk/core/pageWorld.js' in webpack.config.js
2. make a build and install the extension in chrome
3. Open website other than https://mail.google.com and open developer console

**Expected behavior**
It should not throw any such errors if extension is installed with this library.

**Screenshots**
![image](https://user-images.githubusercontent.com/40424777/225303289-34fcc9ac-33c6-46ab-b8f6-2022b48ea153.png)

**Comment by wegry** [comment]
Which version of `@inboxsdk/core` are you seeing this error with?

**Comment by wegry** [comment]
@nirvparekh mind creating a minimum, reproducible example? 

It's not clear to me if the extension in the reproduction steps is intended to be used both within `https://mail.google.com` and on others? If that's the case, we'd be willing to accept a PR.

**Comment by nirvparekh** [comment]
> @nirvparekh mind creating a minimum, reproducible example?
> 
> It's not clear to me if the extension in the reproduction steps is intended to be used both within `https://mail.google.com` and on other domains? If that's the case, we'd be willing to accept a PR.

Thanks @wegry for showing interest in this issue.
Please check this repo: [https://github.com/nirvparekh/inbox-issue-reproduce](https://github.com/nirvparekh/inbox-issue-reproduce), I've created minimal app to reproduce this issue.

There are actually two issues:

1. when it is `https://mail.google.com` :

![image](https://user-images.githubusercontent.com/40424777/231671923-ccf24601-4a6b-4730-ba28-1a6ced29576e.png)

2. other than `https://mail.google.com` :

![image](https://user-images.githubusercontent.com/40424777/231672112-54256264-9075-42f4-a8c9-91f9968157f5.png)

**Comment by wegry** [comment]
Thanks for the repro repo.

For 1:

You might create a separate issue. A quick glance at the minified code shows a `_headerObj`

```sh
$ rg -l -u  "_headersObj"
build/contentScript.js
node_modules/@inboxsdk/core/inboxsdk.js
node_modules/@inboxsdk/core/inboxsdk.js.map
```

This particular issue is present in both `@inboxsdk/core@1.0.0` and (the latest as of writing) `@inboxsdk/core@1.1.3`.

For 2:

It looks like the manifest.json's `contentScripts[0].matches` is set to `["http://*/*", "https://*/*", "<all_urls>"]`. 

The SDK assumes that it'll only be run on `https://mail.google.com`. For example config to get rid of the second error, check out https://github.com/InboxSDK/InboxSDK/blob/c9f3450084488675b40b0104dc22ba052e08e85e/examples/app-menu/manifest.json#L7.

**Comment by nirvparekh** [comment]
@wegry I checked by "matches": ["https://mail.google.com/*"], works fine. no "Should not happen" issue. But that should not be the case because we allow our extension to interact with all websites. we must have to keep like this.
```
"matches": [
  "https://mail.google.com/*",
  "https://*/*"
]
```

**Comment by wegry** [comment]
> we allow our extension to interact with all websites.

@nirvparekh in that case, we'd be willing to accept a PR with SDK configuration to opt out of the logging.

**Comment by nirvparekh** [comment]
@wegry Thank you for reopening.

just FYI, it is error type, not warning type.

**Comment by nirvparekh** [comment]
@wegry  is anyone working to fix this?

**Comment by wegry** [comment]
PRs welcome.

**Comment by wegry** [comment]
This issue was resolved by #875, and that work was released as part of `@inboxsdk/core@1.0.1`.

Just to double check: I can't reproduce this with the SDK just past `@inboxsdk/core@1.2.2`. Here's the combination of things I tried:

- Commit 261337d9
- [a sample extension in this repo](https://github.com/InboxSDK/InboxSDK/blob/261337d96f0a9b98fc5affb86448226399fd3a03/examples/thread-test/content.js#L20
- Navigating to both https://mail.google.com (extension works as expected) and a [toy project site I have](https://brave-chandrasekhar-d23b8b.netlify.app/) (no errors logged)
- and this diff:

```diff
diff --git a/examples/thread-test/manifest.json b/examples/thread-test/manifest.json
index f670be8f..c9c29ce8 100644
--- a/examples/thread-test/manifest.json
+++ b/examples/thread-test/manifest.json
@@ -4,7 +4,7 @@
   "version": "0.1",
   "content_scripts": [
     {
-      "matches": ["https://mail.google.com/*"],
+      "matches": ["https://mail.google.com/*", "https://*/*"],
       "css": [],
       "js": ["inboxsdk.js", "content.js"],
       "run_at": "document_start"
@@ -12,13 +12,13 @@
   ],
   "permissions": ["https://mail.google.com/"],
   "web_accessible_resources": [
-    "inboxsdk.js.map",
-    "pageWorld.js",
-    "pageWorld.js.map"
+    {
+      "resources": ["inboxsdk.js.map", "pageWorld.js", "pageWorld.js.map"],
+      "matches": ["*://*/*"]
+    }
   ],
   "background": {
-    "scripts": ["background.js"],
-    "persistent": false
+    "service_worker": "background.js"
   },
-  "manifest_version": 2
+  "manifest_version": 3
 }
diff --git a/src/platform-implementation-js/platform-implementation.ts b/src/platform-implementation-js/platform-implementation.ts
index 95f7ae43..cee10f91 100644
--- a/src/platform-implementation-js/platform-implementation.ts
+++ b/src/platform-implementation-js/platform-implementation.ts
@@ -194,7 +194,10 @@ export function makePlatformImplementation(
     (process.env.NODE_ENV === 'test' && global.__test_origin) ||
     document.location.origin;
 
+  console.log('Will we resolve?');
+
   if (origin !== 'https://mail.google.com') {
+    console.log('we will not');
     // never resolve
     return new Promise(() => {});
   }
```
